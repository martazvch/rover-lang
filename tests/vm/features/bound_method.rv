struct Person {
    age: int,

    fn speak(self) {
        self.age = self.age * 10
        print self.age
    }

    fn getOld(self, amount: int) {
        self.age = self.age + amount
    }
}

fn oui() {
    print "oui"
}

fn main() {
    var p = Person.{age=5}
    p.speak()

    // Mess up with stack
    var a = 5
    p.speak()
    oui()

    var method1: fn(Person) -> void 
    method1 = p.speak
    var method2: fn(Person) -> void = p.speak
    var method3 = p.speak

    method1()

    p.speak()
    method2()

    p.getOld(8)
    
    method3()
}

