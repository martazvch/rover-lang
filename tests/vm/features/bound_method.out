50
500
"oui"
5000
50000
500000
5000080
